spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/query/StringBasedJdbcQuery.java

```diff
@@ -78,6 +80,8 @@
 public class StringBasedJdbcQuery extends AbstractJdbcQuery {
 
 	private static final String PARAMETER_NEEDS_TO_BE_NAMED = "For queries with named parameters you need to provide names for method parameters; Use @Param for query method parameters, or use the javac flag -parameters";
+	private final static String LOCKING_IS_NOT_SUPPORTED = "Currently, @Lock is supported only on derived queries. In other words, for queries created with @Query, the locking condition specified with @Lock does nothing";
```

Comment:
Would it make sense to reword the second part from a repetition to something actionable? I.e. "…include the database-specific locking hints in your query…"?

Reply:
That makes sense, I agree with you, @mp911de. I'll provide the PR with exception thrown for 4th major release shortly, and I'll apply your suggestion, what do you think?